File: src/Entity/Commande.php
<?php

namespace App\Entity;

use App\Repository\CommandeRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Validator\Context\ExecutionContextInterface;

#[ORM\Entity(repositoryClass: CommandeRepository::class)]
class Commande
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: Types::DATE_MUTABLE, nullable: true)]
    #[Assert\NotBlank(message: "La date ne peut pas être vide.")]
    #[Assert\GreaterThan("today", message: "La date doit être dans le futur.")]
    private ?\DateTimeInterface $date = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "L'adresse ne peut pas être vide.")]
    private ?string $adresse = null;

    /**
     * @var Collection<int, Panier>
     */
    #[ORM\OneToMany(targetEntity: Panier::class, mappedBy: 'commande', cascade: ['persist', 'remove'])]
    private Collection $paniers;

    #[ORM\ManyToOne(inversedBy: 'commandes')]
    private ?Utilisateurs $utilisateurs = null;

    public function __construct()
    {
        $this->paniers = new ArrayCollection();
        $this->date = (new \DateTime())->modify('+1 day'); // Définit la date à demain par défaut
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): static
    {
        $this->date = $date;
        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): static
    {
        $this->adresse = $adresse;
        return $this;
    }

    /**
     * @return Collection<int, Panier>
     */
    public function getPaniers(): Collection
    {
        return $this->paniers;
    }

    public function addPanier(Panier $panier): static
    {
        if (!$this->paniers->contains($panier)) {
            $this->paniers->add($panier);
            $panier->setCommande($this);
        }
        return $this;
    }

    public function removePanier(Panier $panier): static
    {
        if ($this->paniers->removeElement($panier)) {
            if ($panier->getCommande() === $this) {
                $panier->setCommande(null);
            }
        }
        return $this;
    }

    #[Assert\Callback]
    public function validate(ExecutionContextInterface $context): void
    {
        if (empty($this->adresse) && empty($this->date)) {
            $context->buildViolation('Veuillez entrer soit une adresse, soit une date.')
                ->atPath('adresse')
                ->addViolation();
        }
    }

    public function getUtilisateurs(): ?Utilisateurs
    {
        return $this->utilisateurs;
    }

    public function setUtilisateurs(?Utilisateurs $utilisateurs): static
    {
        $this->utilisateurs = $utilisateurs;

        return $this;
    }
}

File: PIDEV-integrationjava/PIDEV-integrationjava/src/main/java/tn/esprit/models/Commande.java
package tn.esprit.models;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class Commande {
    private int id;
    private LocalDate date;
    private String adresse;
    private List<Panier> paniers;
    private Utilisateur utilisateurs;

    public Commande() {
        this.paniers = new ArrayList<>();
        this.date = LocalDate.now().plusDays(1); // Définit la date à demain par défaut
        this.utilisateurs = null; // Initialisation explicite
    }

    public Commande(int id, LocalDate date, String adresse, Utilisateur utilisateurs) {
        this.id = id;
        setDate(date); // Utiliser le setter pour la validation
        setAdresse(adresse); // Utiliser le setter pour la validation
        this.utilisateurs = utilisateurs;
        this.paniers = new ArrayList<>();
    }

    public Commande(LocalDate date, String adresse, Utilisateur utilisateurs) {
        System.out.println("Constructeur Commande appelé - Date : " + date + ", Adresse : '" + adresse + "'");
        setDate(date); // Utilisation du setter pour appliquer les validations
        setAdresse(adresse); // Utiliser le setter pour la validation
        this.utilisateurs = utilisateurs;
        this.paniers = new ArrayList<>();
        System.out.println("Constructeur Commande terminé - Adresse : '" + this.adresse + "', Date : " + this.date);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public LocalDate getDate() {
        System.out.println("getDate appelé - Valeur retournée : " + date);
        return date;
    }

    public void setDate(LocalDate date) {
        System.out.println("setDate appelé avec valeur : " + date);
        if (date == null) {
            System.out.println("Validation échouée : La date est null.");
            throw new IllegalArgumentException("La date ne peut pas être nulle.");
        }
        this.date = date;
        System.out.println("setDate terminé - Date définie : " + this.date);
    }

    public String getAdresse() {
        System.out.println("getAdresse appelé - Valeur retournée : '" + adresse + "'");
        return adresse;
    }

    public void setAdresse(String adresse) {
        System.out.println("setAdresse appelé avec valeur : '" + adresse + "'");
        if (adresse == null || adresse.trim().isEmpty()) {
            System.out.println("Validation échouée : L'adresse est null ou vide.");
            throw new IllegalArgumentException("L'adresse ne peut pas être vide.");
        }
        this.adresse = adresse;
        System.out.println("setAdresse terminé - Adresse définie : '" + this.adresse + "'");
    }

    public List<Panier> getPaniers() {
        return paniers;
    }

    public void setPaniers(List<Panier> paniers) {
        if (paniers == null) {
            this.paniers = new ArrayList<>();
        } else {
            this.paniers = paniers;
        }
    }

    public Utilisateur getUtilisateurs() {
        return utilisateurs;
    }

    public void setUtilisateurs(Utilisateur utilisateurs) {
        this.utilisateurs = utilisateurs;
    }

    public void addPanier(Panier panier) {
        if (panier != null && !this.paniers.contains(panier)) {
            this.paniers.add(panier);
        }
    }

    public void removePanier(Panier panier) {
        this.paniers.remove(panier);
    }

    @Override
    public String toString() {
        return "Commande{" +
                "id=" + id +
                ", date=" + date +
                ", adresse='" + adresse + '\'' +
                ", utilisateur=" + (utilisateurs != null ? utilisateurs.getId_utilisateur() : "null") +
                ", paniers=" + paniers +
                '}';
    }
}